### PR TITLE
Only rerender mobile-sections for wikipedias

### PR DESCRIPTION
**Pull request by Pchelolo**
Bug: https://phabricator.wikimedia.org/T201103

### CODE REVIEW

File: config.example.wikimedia.yaml [hunk 1]
```diff
@@ -275,12 +271,32 @@ spec: &spec
...
-                        uri: 'https://{{message.meta.domain}}/api/rest_v1/page/mobile-sections/{decode(match.meta.uri.title)}'
-                        headers:
-                          cache-control: no-cache
-                        query:
-                          redirect: false
+                  - match:
+                      meta:
+                        uri: '/^(?<proto>https?):\/\/[^\/]+\/wiki\/(?<title>.+)$/'
+                      tags:
+                      - purge
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: indent one more level

File: config.example.wikimedia.yaml [hunk 2]
```diff
@@ -275,12 +271,32 @@ spec: &spec
...
+                    match_not:
+                      meta:
+                        domain: '/(?:.*\.wikipedia\.org$)|(?:.*\.wiktionary\.org$)/'
+                    exec: &mw_purge_others_rerender
+                    - method: get
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: idem

File: config.example.wikimedia.yaml [hunk 3]
```diff
@@ -275,12 +271,32 @@ spec: &spec
...
-                        query:
-                          redirect: false
+                  - match:
+                      meta:
+                        uri: '/^(?<proto>https?):\/\/[^\/]+\/wiki\/(?<title>.+)$/'
+                      tags:
+                      - purge
+                    match_not:
+                      meta:
+                        domain: '/(?:.*\.wikipedia\.org$)|(?:.*\.wiktionary\.org$)/'
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: wikidata.org can be skipped here as well

File: config.example.wikimedia.yaml [hunk 4]
```diff
@@ -290,35 +306,30 @@ spec: &spec
...
+                    - purge
+                  exec: *mw_purge_wiktionary_rerender
+                - match:
+                    meta:
+                      uri: '/^(?<proto>https?):\/\/[^\/]+\/wiki\/(?<title>.+)$/'
+                    tags:
+                    - purge
+                  match_not:
+                    meta:
+                      domain: '/(?:.*\.wikipedia\.org$)|(?:.*\.wiktionary\.org$)/'
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: idem, can exclude wikidata too

File: config.example.wikimedia.yaml [hunk 5]
```diff
@@ -290,35 +306,30 @@ spec: &spec
                     - 412
                 limiters:
                   blacklist: 'html:{message.meta.uri}'
-                match:
-                  meta:
-                    uri: '/^(?<proto>https?):\/\/[^\/]+\/wiki\/(?<title>.+)$/'
-                  tags:
+                cases:
+                - match:
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: indent

File: config.example.wikimedia.yaml [hunk 6]
```diff
@@ -581,70 +592,100 @@ spec: &spec
...
                           cache-control: no-cache
                         query:
                           redirect: false
+                  - match:
+                      meta:
+                        uri: '/https:\/\/[^\/]+\/wiki\/(?<title>.+)/'
+                      tags: [ 'wikidata' ]
+                    match_not:
+                      meta:
+                        domain: '/(?:.*\.wikipedia\.org$)|(?:.*\.wiktionary\.org$)/'
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: add wikidata.org

File: config.example.wikimedia.yaml [hunk 7]
```diff
@@ -581,70 +592,100 @@ spec: &spec
...
+                    exec:
+                    - method: get
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: indent

File: config.example.wikimedia.yaml [hunk 8]
```diff
@@ -581,70 +592,100 @@ spec: &spec
...
+                topic: mediawiki.page-properties-change
+                # We don't support 'OR' in the match section, so workaround it by 2 cases with identical exec
+                cases:
+                  - match:
+                      meta:
+                        domain: '/.*\.wikipedia.org$/'
+                      added_properties:
+                        page_image: '/.+/'
+                    exec: &page_images_wikipedia_rerender
+                    - method: get
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: indent

File: config.example.wikimedia.yaml [hunk 9]
```diff
@@ -581,70 +592,100 @@ spec: &spec
...
+                      - meta:
+                         domain: '/(?:.*\.wikipedia\.org$)|(?:.*\.wiktionary\.org$)/'
+                      - meta:
+                          domain: /\.wikidata\.org$/
+                        page_namespace: 0
+                      - meta:
+                          domain: /\.wikidata\.org$/
+                        page_namespace: 120
+                    exec: &page_images_others_rerender
+                    - method: get
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: indent

File: config.example.wikimedia.yaml [hunk 10]
```diff
@@ -581,70 +592,100 @@ spec: &spec
...
                       meta:
+                        domain: '/.*\.wikipedia.org$/'
+                      removed_properties:
+                        page_image: '/.+/'
+                    exec: *page_images_wikipedia_rerender
+                  - match:
+                      removed_properties:
+                        page_image: '/.+/'
+                    match_not:
+                    - meta:
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: indent

File: config.example.wikimedia.yaml [hunk 11]
```diff
@@ -581,70 +592,100 @@ spec: &spec
...
+                        domain: '/(?:.*\.wikipedia\.org$)|(?:.*\.wiktionary\.org$)/'
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: can exclude wd.org

**Comment by Pchelolo** [comment]
Gave it some love

File: config.example.wikimedia.yaml [hunk 12]
```diff
@@ -581,58 +598,48 @@ spec: &spec
...
-                          domain: /\.wikidata\.org$/
-                        page_namespace: 120
+                      meta:
+                        uri: '/https:\/\/[^\/]+\/wiki\/(?<title>.+)/'
+                        domain: '/.*\.wikipedia.org$/'
+                      tags: [ 'wikidata' ]
                     exec:
                       - method: get
-                        uri: 'https://{{message.meta.domain}}/api/rest_v1/page/summary/{message.page_title}'
+                        uri: 'https://{{message.meta.domain}}/api/rest_v1/page/summary/{{match.meta.uri.title}}'
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: `decode(match.meta.uri/title)` ?

File: config.example.wikimedia.yaml [hunk 13]
```diff
@@ -581,58 +598,48 @@ spec: &spec
...
                         headers:
                           cache-control: no-cache
                         query:
                           redirect: false
                       - method: get
-                        uri: 'https://{{message.meta.domain}}/api/rest_v1/page/mobile-sections/{message.page_title}'
+                        uri: 'https://{{message.meta.domain}}/api/rest_v1/page/mobile-sections/{{match.meta.uri.title}}'
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: Idem

File: config.example.wikimedia.yaml [hunk 14]
```diff
@@ -581,58 +598,48 @@ spec: &spec
...
-                        page_image: '/.+/' # Regex that matches anything just to check the prop is set
-                    match_not:
                       meta:
-                        domain: /\.wikidata\.org$/
+                        uri: '/https:\/\/[^\/]+\/wiki\/(?<title>.+)/'
+                      tags: [ 'wikidata' ]
+                    match_not: *others_match_not
                     exec:
+                      - method: get
+                        uri: 'https://{{message.meta.domain}}/api/rest_v1/page/summary/{{match.meta.uri.title}}'
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: Idem

File: config.example.wikimedia.yaml [hunk 15]
```diff
@@ -645,6 +652,28 @@ spec: &spec
                           cache-control: no-cache
                         query:
                           redirect: false
+                  - match:
+                      removed_properties:
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: `s/removed/added/`

File: config.example.wikimedia.yaml [hunk 16]
```diff
@@ -645,6 +652,28 @@ spec: &spec
...
+                        page_image: '/.+/'
+                    match_not: *others_match_not
+                    exec: &page_images_others_rerender
+                    - method: get
```
Comment (d00rman) [inline config.example.wikimedia.yaml]: Indent

File: config.example.wikimedia.yaml (hunk 12 above)
Comment (Pchelolo) [reply config.example.wikimedia.yaml]: nono... this is an incorrect diff actually. the left one belongs to a different rule, so this one is solid.

File: config.example.wikimedia.yaml (hunk 13 above)
Comment (Pchelolo) [reply config.example.wikimedia.yaml]: Idem

File: config.example.wikimedia.yaml (hunk 14 above)
Comment (Pchelolo) [reply config.example.wikimedia.yaml]: Idem

**Pull request merged**